src/composables/useTransactions.ts

```diff
@@ -307,17 +302,12 @@ export default function useTransactions() {
       txs.forEach((tx, txIndex) => {
         if (tx != null) {
           finalizeTransaction(tx.transactionHash, 'tx', tx);
-          shouldRefetchBalances = true;
         } else {
           updateTransaction(pendingTxActivity.value[txIndex].id, 'tx', {
             lastCheckedBlockNumber: blockNumber.value
           });
         }
       });
-
-      if (shouldRefetchBalances) {
-        refetchBalances.value();
-      }
```

Comment:
This is related to Gnosis as well - essentially blocknative is able to detect when those trades are settled and is able to refetch balances.